src/Expander/Monad.hs

```diff
@@ -267,9 +268,10 @@ data ExpanderState = ExpanderState
   , _expanderTasks :: [(TaskID, ExpanderLocal, ExpanderTask)]
   , _expanderOriginLocations :: !(Map.Map SplitCorePtr SrcLoc)
   , _expanderCompletedCore :: !(Map.Map SplitCorePtr (CoreF TypePatternPtr PatternPtr SplitCorePtr))
-  , _expanderCompletedPatterns :: !(Map.Map PatternPtr ConstructorPattern)
+  , _expanderCompletedPatterns :: !(Map.Map PatternPtr (ConstructorPatternF PatternPtr))
```

Comment:
At some point, we should probably switch to a weak map, we're probably keeping data in those `Map`s long after they are no longer needed...

Reply:
Don't we clear out those tables between modules? If we don't, then we should.

Reply:
ah, that's much simpler than using week references, good idea.